nntrainer/layers/layer_node.cpp

```diff
@@ -129,21 +129,73 @@ class SharedFrom : public Name {
  *
  */
 LayerNode::~LayerNode() = default;
+  
+/**
+ * @brief get the compute engine property from property string vector
+ *  : default is CPU
+ * @return LayerComputeEngine Enum : CPU, GPU, QNN
+ *
+ */
+ml::train::LayerComputeEngine
+getComputeEngine(const std::vector<std::string> &props) {
+  for (auto &prop : props) {
+    std::string key, value;
+    int status = nntrainer::getKeyValue(prop, key, value);
+    if (nntrainer::istrequal(key, "engine")) {
+      constexpr const auto data =
+        std::data(props::ComputeEngineTypeInfo::EnumList);
+      for (uint i = 0; i < props::ComputeEngineTypeInfo::EnumList.size(); ++i) {
+        if (nntrainer::istrequal(value.c_str(),
+                                 props::ComputeEngineTypeInfo::EnumStr[i])) {
+          return data[i];
+        }
+      }
+    }
+  }
+
+  return ml::train::LayerComputeEngine::CPU;
+}
+
+/**
+ * @brief get the compute engine property from property string vector
+ *  : default is CPU
+ * @return LayerComputeEngine Enum : CPU, GPU, QNN
+ *
+ */
+ml::train::LayerComputeEngine
+getComputeEngine(const std::vector<std::string> &props) {
+  for (auto &prop : props) {
+    std::string key, value;
+    int status = nntrainer::getKeyValue(prop, key, value);
+    if (nntrainer::istrequal(key, "engine")) {
+      if (nntrainer::istrequal(value, "qnn")) {
+        return ml::train::LayerComputeEngine::QNN;
+      } else if (nntrainer::istrequal(value, "gpu")) {
+        return ml::train::LayerComputeEngine::GPU;
+      }
+    }
+  }
+
+  return ml::train::LayerComputeEngine::CPU;
+}
 
 /**
  * @brief Layer factory creator with constructor
  */
 std::unique_ptr<LayerNode>
 createLayerNode(const ml::train::LayerType &type,
-                const std::vector<std::string> &properties,
-                const ml::train::LayerComputeEngine &compute_engine) {
+                const std::vector<std::string> &properties) {
+
+  if (getComputeEngine(properties) == ml::train::LayerComputeEngine::GPU) {
 #ifdef ENABLE_OPENCL
-  if (compute_engine == ml::train::LayerComputeEngine::GPU) {
     auto &cc = nntrainer::ClContext::Global();
-    return createLayerNode(cc.createObject<nntrainer::Layer>(type), properties,
-                           compute_engine);
-  }
+    return createLayerNode(cc.createObject<nntrainer::Layer>(type), properties);
+#else
+    throw std::invalid_argument(
+      "opencl layer creation without enable-opencl option");
 #endif
+  }
+
   auto &ac = nntrainer::AppContext::Global();
```

Comment:
Do you think it holds for both `ml::train::LayerComputeEngine::CPU` and `ml::train::LayerComputeEngine::QNN`?
If not, you may need to add more condition to check it.
I think it would be better to add a case to check `QNN` with NYI exception.